src/DoctrineDBALBundle.php

```diff
@@ -22,7 +12,30 @@ class DoctrineDBALBundle extends Bundle
     /**
      * {@inheritDoc}
      */
-    public function registerCommands(Application $application)
+    public function build(ContainerBuilder $container)
     {
+        parent::build($container);
+        $container->addCompilerPass(new WellKnownSchemaFilterPass());
+        $container->addCompilerPass(new DbalSchemaFilterPass());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function shutdown()
+    {
+        // Close all connections to avoid reaching too many connections in the process when booting again later (tests)
+        if (! $this->container->hasParameter('doctrine.connections')) {
+            return;
+        }
+
+        // TODO: use ConnectionRegistry?
```

Comment:
no, because this would not let us avoid closing uninitialized connections.

What we could use instead is an IteratorArgument using "weak" references, but that's not worth it here IMO.